shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/rewrite/token/IndexTokenGeneratorTest.java

```diff
@@ -0,0 +1,80 @@
+import org.apache.shardingsphere.sharding.rule.ShardingRule;
+import org.apache.shardingsphere.sql.parser.sql.common.segment.ddl.index.IndexSegment;
+import org.apache.shardingsphere.sql.parser.sql.common.value.identifier.IdentifierValue;
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public final class IndexTokenGeneratorTest {
+
+    private static final int TEST_START_INDEX = 1;
+
+    private static final int TEST_STOP_INDEX = 3;
+
+    @Test
+    public void assertIsGenerateSQLToken() {
+        CreateDatabaseStatementContext createDatabaseStatementContext = mock(CreateDatabaseStatementContext.class);
+        IndexTokenGenerator indexTokenGenerator = new IndexTokenGenerator();
+        assertFalse(indexTokenGenerator.isGenerateSQLToken(createDatabaseStatementContext));
+        AlterIndexStatementContext alterIndexStatementContext = mock(AlterIndexStatementContext.class);
+        Collection<IndexSegment> indexSegmentCollection = new LinkedList<>();
+        when(alterIndexStatementContext.getIndexes()).thenReturn(indexSegmentCollection);
+        assertFalse(indexTokenGenerator.isGenerateSQLToken(alterIndexStatementContext));
+        indexSegmentCollection.add(mock(IndexSegment.class));
+        assertTrue(indexTokenGenerator.isGenerateSQLToken(alterIndexStatementContext));
+    }
+
+    @Test
+    public void assertGenerateSQLTokens() {
+        IndexSegment indexSegment = mock(IndexSegment.class);
+        when(indexSegment.getStartIndex()).thenReturn(TEST_START_INDEX);
+        when(indexSegment.getStopIndex()).thenReturn(TEST_STOP_INDEX);
+        IdentifierValue identifierValue = mock(IdentifierValue.class);
+        when(indexSegment.getIdentifier()).thenReturn(identifierValue);
+        Collection<IndexSegment> indexSegmentCollection = new LinkedList<>();
+        indexSegmentCollection.add(indexSegment);
+        AlterIndexStatementContext alterIndexStatementContext = mock(AlterIndexStatementContext.class);
+        when(alterIndexStatementContext.getIndexes()).thenReturn(indexSegmentCollection);
+        IndexTokenGenerator indexTokenGenerator = new IndexTokenGenerator();
+        ShardingRule shardingRule = mock(ShardingRule.class);
+        indexTokenGenerator.setShardingRule(shardingRule);
+        ShardingSphereSchema schema = mock(ShardingSphereSchema.class);
+        indexTokenGenerator.setSchema(schema);
+        Collection<IndexToken> result = indexTokenGenerator.generateSQLTokens(alterIndexStatementContext);
+        assertThat(result.size(), is(1));
+        assertThat(result.stream().collect(Collectors.toList()).get(0).getStartIndex(), is(TEST_START_INDEX));
```

Comment:
`result.stream().collect(Collectors.toList())` can be replaced with `new ArrayList()`